### ISSUE TITLE
i6, 9.0.2 ftp0 enabled (cl0ver), nvram: Error setting variable - 'com.apple.System.boot-nonce'

### ISSUE THREAD

**Issue by JustasL** [tihmstar/futurerestore#62]
Hello, I am having trouble setting com.apple.System.boot-nonce to shsh2 generator.

I am using iPhone 6 with 9.0.2 JB, I've enabled ftp0 with cl0ver, I did run /nonceEnabler, which in the end said:

[*] patching bytes at=0xffffff80137cbcd8
[*] done patching

I guess it means it was a success, but I can't run nvram com.apple.System.boot-nonce=MY_GENERATOR

I've tried doing this:
>turn your phone to recovery mode. reset nvram by typing: "irecovery -r". then "irecovery -s" and type: "setenv auto-boot=true","saveenv" and "reboot".
As suggested here: https://www.reddit.com/r/jailbreak/comments/5ladq5/discussion_futurerestore_has_been_updated/dbuasjt/

But it didn't helped.
Any suggestions, ideas?

**Comment by JustasL** [comment]
Found a solution mentioned in reddit comments here: 
https://www.reddit.com/r/jailbreak/comments/5tmi4l/help_nvram_error_setting_variable_while_trying_to/ddxovpb/

>TL;DR: nonceEnabler doesn't always work, use nvpatch from kern-utils instead.

Can confirm, it does fix the bug

**Comment by brymonster** [comment]
how did u get the nonce set with nvpatch?

**Comment by brymonster** [comment]
Thanks! I got it to patch successfully finally. 
But my nvram -p seems weird. Seems the nonce I have had set finest change. Idk if 
that good or bad. Still didn't work tho. 
https://github.com/Siguza/cl0ver/issues/30#issuecomment-282987561

-bryan

**Comment by brymonster** [comment]
@tihmstar 
like @JustasL my device was stuck `waiting for device..` 
does this mean my generator is not properly set? 
i have tried mulitple times with nonceEnabler and ios-kern nvpatch
still the same:
>nvram -p
backlight-level	1607
com.apple.System.fp-state	%00%00%00%00`%00%00%00%00%00%00%00%00%00%00%00%00%00%00%00
boot-args	
auto-boot	true
com.apple.system.boot-nonce	0x16720224aed2294e

**Comment by tihmstar** [comment]
This isn't really futurerestore related
